### ISSUE TITLE
event_revoker ignores errors while manual unregistration does not

### ISSUE THREAD

**Issue by sylveon** [microsoft/cppwinrt#1003]
As can be seen here, the `hresult` return is ignored:

https://github.com/microsoft/cppwinrt/blob/7e730e7652a42754bd0384feb7c84d5484daff2e/strings/base_events.h#L57-L63

However, in the code generated for manual unregistration, the return value is asserted to be S_OK:
https://github.com/microsoft/cppwinrt/blob/4366357db991bda7bb615dedabdb3637cc93e9e5/cppwinrt/code_writers.h#L1121-L1128

This means that the behavior is different: I was migrating code to use manual unregistration because it was already holding a strong reference so storing a single `event_token` takes much less storage. While `event_revoker` worked fine, manual unregistration triggers the assert if the RPC server has died while trying to unregister the callback (because then the method to unregister the callback returns `RPC_S_SERVER_UNAVAILABLE`).

**Comment by kennykerr** [comment]
Yep, the event revoker came much later and by then I realized there was little point in propagating errors resulting from unregistering an event. Ideally we could make the manual unregistration behave the same way. I can't imagine anyone depending on this, but you never know.

**Comment by sylveon** [comment]
People have probably inadvertently started depending on event_revoker ignoring errors (as I did, until I tried manual unregistration), and currently manual unregistration errors are a fatal failure in debug mode and ignored in release mode.

So I don't think anybody depends on asserts firing (because they terminate, and sudden termination is something nobody really depends on), and the "correct" behavior already exists in release mode. Ignoring errors in manual unregistration will bring consistency and should not result in any breakages.
